dask/highlevelgraph.py

```diff
@@ -457,6 +478,67 @@ def cull(self, keys: Set):
 
         return HighLevelGraph(ret_layers, ret_dependencies, ret_key_deps)
 
+    def map_basic_layers(
```

Comment:
API question: What do you think about having a more generic `map_layers` method where the `func` being mapped is responsible for checking the layer type? This would be to prevent us from needing to add `map_*_layers` methods as the number of layer types grows. 

Apologies for not bringing this point up earlier in the process

Reply:
My sense here is that this probably makes sense, but that it also probably doesn't need to be handled right now.  The API choices here don't preclude us adding intermediate methods like `map_layers` later.

Reply:
Sounds good, we can revisit this point later on as needed 👍

Reply:
> API question: What do you think about having a more generic `map_layers` method where the `func` being mapped is responsible for checking the layer type? This would be to prevent us from needing to add `map_*_layers` methods as the number of layer types grows.

Make sense, my thinking behind `map_basic_layers()` was to have a map for the layers that are materialized low level graphs. And if the user wants to apply something on all layers, it is always possible to access the layers of a HLG directly. 

But I agree, let's wait a bit before we settle on the API.